src/middleware.ts

```diff
@@ -0,0 +1,42 @@
+import { sequence, defineMiddleware } from 'astro:middleware';
+import { jwtVerify, importSPKI } from 'jose';
+
+const auth = defineMiddleware(async ({ locals, request }, next) => {
+	// Get the Authorization header.
+	const authHeader = request.headers.get('Authorization');
+	if (!authHeader) {
+		// If it's not present, the user is not authenticated.
+		locals.userId = null;
+		locals.isAuthenticated = false;
+		return next();
+	}
+
+	// Remove the "Bearer " prefix to get the raw token.
+	const rawToken = authHeader.replace('Bearer ', '');
+
+	// Import the public key.
+	const publicKey = await importSPKI(import.meta.env.JWT_PUBLIC_KEY, 'EdDSA');
+
+	// Verify the token.
+	let payload;
+	try {
+		({ payload } = await jwtVerify(rawToken, publicKey, {
+			algorithms: ['EdDSA'],
+			clockTolerance: 5,
+			requiredClaims: ['sub'],
+		}));
+	} catch (e) {
+		// If the token is invalid, the user is not authenticated.
+		locals.userId = null;
+		locals.isAuthenticated = false;
+		return next();
+	}
+
+	// If the token is valid, the user is authenticated.
+	locals.authHeader = authHeader ?? undefined;
+	locals.userId = payload.sub ?? null;
+	locals.isAuthenticated = payload.sub !== undefined;
+	return next();
```

Comment:
Can be moved inside try catch
```suggestion
	try {
		const { payload } = await jwtVerify(rawToken, publicKey, {
			algorithms: ['EdDSA'],
			clockTolerance: 5,
			requiredClaims: ['sub'],
		});
              locals.authHeader = authHeader ?? undefined;
             	locals.userId = payload.sub ?? null;
             	locals.isAuthenticated = payload.sub !== undefined;
              return next();
	} catch (e) {
		// If the token is invalid, the user is not authenticated.
		locals.userId = null;
		locals.isAuthenticated = false;
		return next();
	}
```

Reply:
Yeah, that's cleaner.